### PR TITLE
Build istio-iptables into istio-cni.

### PR DESCRIPTION
**Pull request by bianpengyuan**
For #32901. Also add a file output path param to iptables, which will be used for CNI dry run test in later PRs.

### CODE REVIEW

File: cni/pkg/plugin/iptables.go [hunk 1]
```diff
@@ -35,32 +38,28 @@ func newIPTables() InterceptRuleMgr {
...
-		"-m", rdrct.redirectMode,
-		"-i", rdrct.includeIPCidrs,
-		"-b", rdrct.includePorts,
-		"-d", rdrct.excludeInboundPorts,
-		"-o", rdrct.excludeOutboundPorts,
-		"-x", rdrct.excludeIPCidrs,
-		"-k", rdrct.kubevirtInterfaces,
-	}
+	viper.Set(constants.CNIMode, true)
+	viper.Set(constants.NetworkNamespace, netns)
```
Comment (howardjohn) [inline cni/pkg/plugin/iptables.go:42]: is this safe to mutate? Seems like we are mutating global variables from within normal function. Does CNI call this multiple times? Or even worse, in parallel?

File: tools/istio-iptables/pkg/validation/validator.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/prometheus/common/log"
```
Comment (howardjohn) [inline tools/istio-iptables/pkg/validation/validator.go:26]: wrong log package :slightly_smiling_face:

File: tools/istio-iptables/pkg/dependencies/implementation.go [hunk 3]
```diff
@@ -60,12 +61,24 @@ var XTablesCmds = sets.NewSet(
...
+	log.Infof("%s %s\n", cmd, strings.Join(args, " "))
+	if r.CNIMode {
+		originalCmd := cmd
+		cmd = constants.NSENTER
+		args = append([]string{fmt.Sprintf("--net=%v", r.NetworkNamespace), "--", originalCmd}, args...)
+	}
 	externalCommand := exec.Command(cmd, args...)
-	externalCommand.Stdout = os.Stdout
+	if r.CNIMode {
+		externalCommand.Stdout = os.Stderr
```
Comment (howardjohn) [inline tools/istio-iptables/pkg/dependencies/implementation.go:78]: Its not clear to me why its different, could use a comment

**Review by howardjohn — COMMENTED** [review]
I know it is marked WIP but overall LGTM :+1:

File: tools/istio-iptables/pkg/validation/validator.go (hunk 2 above)
Comment (bianpengyuan) [reply tools/istio-iptables/pkg/validation/validator.go]: Thanks! This is unrelated changes to this PR. Removed.

File: cni/pkg/plugin/iptables.go (hunk 1 above)
Comment (bianpengyuan) [reply cni/pkg/plugin/iptables.go:40]: Yeah, for every new pod, kubelet will exec a new CNI network plugin process. This function will only be executed once per process.

File: tools/istio-iptables/pkg/dependencies/implementation.go (hunk 3 above)
Comment (bianpengyuan) [reply tools/istio-iptables/pkg/dependencies/implementation.go]: Sorry it is debug left over. I was trying to figure out why CNI network configuration fails during development.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**